test/functional/inputs_test.rb

```diff
@@ -67,6 +74,56 @@
     end
   end
 
+  describe "when using the --input inline raw input flag CLI option" do
+    let(:result) { run_inspec_process("exec #{inputs_profiles_path}/cli #{input_opt} #{control_opt}", json: true) }
+    let(:control_opt) { "" }
+
+    describe "when the --input is used once with one value" do
+      let(:input_opt) { "--input test_input_01=value_from_cli_01" }
+      let(:control_opt) { "--controls test_control_01" }
```

Comment:
These don't really need to be functional tests. Processing the command line args directly and then making sure that the options are processed correctly would suffice and cost nothing in comparison to this very heavyweight alternative.

Reply:
OK, I cut an issue for that: https://github.com/inspec/inspec/issues/4412